### ISSUE TITLE
Inheriting from a generic class may not work

### ISSUE THREAD

**Issue by or150** [dotnet/roslyn#7647]
There is a scenario, in which a non-generic class is not able to derive from a generic class.

Assume you have a set of abstract classes defined like this:
```C#
    //Abstract non generic type
    public abstract class A
    {
        public abstract bool Do(object obj);
    }

    //Abstract generic type that exposes generic overloads to A
    public abstract class A<T> : A
    {
        public override bool Do(object obj)
        {
            if (obj is T)
                return Do((T)obj);
            return false;
        }
        public abstract bool Do(T obj);
    }

    public abstract class B<T> : A<T>
    {
        public override bool Do(T obj)
        {
            //some complex code here
            return TypeSpecificLogic(obj) == 12;
        }

        protected abstract int TypeSpecificLogic(T obj);
    }
```

The following code won't compile, even tough the old csc would compile it:
```C#
    public class C : B<object>
    {
        protected override int TypeSpecificLogic(object obj)
        {
            return 12;
        }
    }
```

The error produced is:
`Error CS0534: 'C' does not implement inherited abstract member 'A<object>.Do(object)'`

**Comment by leppie** [comment]
Works for me. Anything specific to repro?

**Comment by Suchiman** [comment]
@leppie this is weird... i pasted the code into one .NET 4.5.2 MVC web application i had open and encountered the same error. I then created a new .NET 4.5.2 console application and pasted that code there too and it didn't reproduce.

**Comment by leppie** [comment]
Was the error for build or intellisense? To check if anyone encounters this again. Could have also just been caused be leftover temporary ASP.NET assemblies.

**Comment by Suchiman** [comment]
It was for the build. IntelliSense showed nothing and after rebuild the error list flashed for a second with build failed but contained nothing. The output window then showed one error, the one named here.

**Comment by or150** [comment]
I just built a ConsoleApplication\Library with the code specified above using vs2015.
I decided to build it against the nuget packages, and looks like it was solved somewhere in 1.1.0.

**Comment by gafter** [comment]
Closing as not reproducible in 1.1 (VS2015 Update 1)
